frontend/src/metabase/query_builder/components/NativeQueryEditor/DataSourceSelectors/DataSourceSelectors.jsx

```diff
@@ -58,12 +60,26 @@ const DataSourceSelectors = ({
   setDatabaseId,
   setTableId,
   requireWriteback = false,
+  editorContext,
 }) => {
   const database = query.database();
-  const databases = query.metadata().databasesList({ savedQuestions: false });
+
+  const databases = useMemo(() => {
+    const allDatabases = query
+      .metadata()
+      .databasesList({ savedQuestions: false });
+
+    if (!requireWriteback) {
+      return allDatabases;
+    }
+
+    return allDatabases.filter(database =>
+      checkDatabaseActionsEnabled(database),
```

Comment:
`database.hasActionsEnabled()`

Reply:
so, a weird consequence of this filter behavior for non-admin users is we let them view queries for databases that they don't have access to. This is probably a minor correctness issue, but if you have access to a write-enabled DB, and view a query action for another write-enabled db, the action editor will incorrectly show you the name of the DB you have access to.

Example:

Admin | limited user
--|--
![Screen Shot 2023-03-13 at 10 07 12 AM](https://user-images.githubusercontent.com/30528226/224759694-b5c93dfd-7eab-46fe-a295-508b3ed0a3f0.png) | ![Screen Shot 2023-03-13 at 10 07 18 AM](https://user-images.githubusercontent.com/30528226/224759699-5d9aeefc-99ef-4a9f-9284-cc34358a3d61.png)